homeassistant/components/myq/light.py

```
@@ -0,0 +1,132 @@
+        """Initialize with API object, device id."""
+        super().__init__(coordinator)
+        self._device = device
+
+    @property
+    def name(self):
+        """Return the name of the garage door if any."""
+        return self._device.name
+
+    @property
+    def available(self):
+        """Return if the device is online."""
+        if not self.coordinator.last_update_success:
+            return False
+
+        # Not all devices report online so assume True if its missing
+        return self._device.device_json[MYQ_DEVICE_STATE].get(
+            MYQ_DEVICE_STATE_ONLINE, True
+        )
+
+    @property
+    def is_on(self):
+        """Return true if the light is on, else False."""
+        return MYQ_TO_HASS.get(self._device.state) == STATE_ON
+
+    @property
+    def is_off(self):
+        """Return true if the light is off, else False."""
+        return MYQ_TO_HASS.get(self._device.state) == STATE_OFF
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return 0
+
+    @property
+    def unique_id(self):
+        """Return a unique, Home Assistant friendly identifier for this entity."""
+        return self._device.device_id
+
+    async def async_turn_on(self, **kwargs):
+        """Issue on command to light."""
+        if self.is_on:
+            return
+
+        try:
+            await self._device.turnon(wait_for_state=True)
+        except MyQError as err:
+            _LOGGER.error(
+                "Turning light %s on failed with error: %s", self._device.name, str(err)
+            )
+
+            return
+
+        # Write new state to HASS
+        self.async_write_ha_state()
+
+    async def async_turn_off(self, **kwargs):
+        """Issue off command to light."""
+        if self.is_off:
+            return
+
+        try:
+            await self._device.turnoff(wait_for_state=True)
+        except MyQError as err:
+            _LOGGER.error(
```

If you wrap this in `HomeAssistantError` and let it raise, the user will get a toast in the UI if this fails